planner/core/task.go

```diff
@@ -1417,7 +1421,11 @@ func BuildFinalModeAggregation(
 				}
 			}
 
-			finalAggFunc.Mode = aggregation.FinalMode
+			if aggFunc.Mode == aggregation.CompleteMode {
```

Comment:
I do not understand what does this code block mean

Reply:
In the old assumption, **before the coprocessor push down, the agg mode is either Complete or Final**. So after the push down, the parent agg become Final mode.

But in the new assumption, before the coprocessor push down, the agg mode can be partial. So here the parent agg mode is set to different value accordingly.

For final before, we set it to final.
For partial, it's set to partial2